app/calendar/edit-habit.tsx

```diff
@@ -0,0 +1,97 @@
+import { useSWRConfig, Fetcher } from "swr";
+import { useState } from "react";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useSession } from "next-auth/react";
+import { Dispatch, SetStateAction } from "react";
+
+export default function EditHabit({
+  habit,
+  setCurrentHabit,
+}: {
+  habit: string;
+  setCurrentHabit: Dispatch<SetStateAction<string>>;
+}) {
+  const [newHabitName, editHabitName] = useState(habit);
```

Comment:
```suggestion
  const [newHabitName, setNewHabitName] = useState(habit);
```
Better to follow the naming convention:
> The convention is to name state variables like [something, setSomething] using [array destructuring.](https://javascript.info/destructuring-assignment)

source: [react docs](https://react.dev/reference/react/useState#usage)